### ISSUE TITLE
Please update go to 1.21.9 or 1.22.2 for CVE GHSA-4v7x-pqxf-cx7m

### ISSUE THREAD

**Issue by akoshok1** [Kong/kubernetes-ingress-controller#5900]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

CVE [GHSA-4v7x-pqxf-cx7m](https://vat.dso.mil/vat/finding?name=GHSA-4v7x-pqxf-cx7m) shows up in scans against 3.1.3

### Expected Behavior

Ideally this CVE will not show up in scans.

### Steps To Reproduce

```markdown
Scan deck 3.1.3 with Anchore, TwistLock or the like.  Verify CVE GHSA-4v7x-pqxf-cx7m
```

### Kong Ingress Controller version

```shell
3.1.3
```

### Kubernetes version

```shell
Any version
```

**Comment by czeslavo** [comment]
https://pkg.go.dev/vuln/GO-2024-2687 for context.

**Comment by pmalek** [comment]
KIC 3.1.5 has been released and it already uses go 1.21.9.

KIC 3.2.0 will be released probably using 1.22.4.
